tools/src/main/java/org/apache/kafka/tools/reassign/ReassignPartitionsCommand.java

```diff
@@ -0,0 +1,1501 @@
+     *
+     * @return                      A result that is useful for testing.
+     */
+    static VerifyAssignmentResult verifyAssignment(Admin adminClient, String jsonString, Boolean preserveThrottles) throws ExecutionException, InterruptedException, JsonProcessingException {
+        Tuple2<List<Tuple2<TopicPartition, List<Integer>>>, Map<TopicPartitionReplica, String>> t0 = parsePartitionReassignmentData(jsonString);
+
+        List<Tuple2<TopicPartition, List<Integer>>> targetParts = t0.v1;
+        Map<TopicPartitionReplica, String> targetLogDirs = t0.v2;
+
+        Tuple2<Map<TopicPartition, PartitionReassignmentState>, Boolean> t1 = verifyPartitionAssignments(adminClient, targetParts);
+
+        Map<TopicPartition, PartitionReassignmentState> partStates = t1.v1;
+        Boolean partsOngoing = t1.v2;
+
+        Tuple2<Map<TopicPartitionReplica, LogDirMoveState>, Boolean> t2 = verifyReplicaMoves(adminClient, targetLogDirs);
+
+        Map<TopicPartitionReplica, LogDirMoveState> moveStates = t2.v1;
+        Boolean movesOngoing = t2.v2;
+
+        if (!partsOngoing && !movesOngoing && !preserveThrottles) {
+            // If the partition assignments and replica assignments are done, clear any throttles
+            // that were set.  We have to clear all throttles, because we don't have enough
+            // information to know all of the source brokers that might have been involved in the
+            // previous reassignments.
+            clearAllThrottles(adminClient, targetParts);
+        }
+
+        return new VerifyAssignmentResult(partStates, partsOngoing, moveStates, movesOngoing);
+    }
+
+    /**
+     * Verify the partition reassignments specified by the user.
+     *
+     * @param adminClient           The AdminClient to use.
+     * @param targets               The partition reassignments specified by the user.
+     *
+     * @return                      A tuple of the partition reassignment states, and a
+     *                              boolean which is true if there are no ongoing
+     *                              reassignments (including reassignments not described
+     *                              in the JSON file.)
+     */
+    private static Tuple2<Map<TopicPartition, PartitionReassignmentState>, Boolean> verifyPartitionAssignments(Admin adminClient,
+                                                                                                               List<Tuple2<TopicPartition, List<Integer>>> targets) throws ExecutionException, InterruptedException {
+        Tuple2<Map<TopicPartition, PartitionReassignmentState>, Boolean> t0 = findPartitionReassignmentStates(adminClient, targets);
+
+        Map<TopicPartition, PartitionReassignmentState> partStates = t0.v1;
+        Boolean partsOngoing = t0.v2;
+
+        System.out.println(partitionReassignmentStatesToString(partStates));
+        return new Tuple2<>(partStates, partsOngoing);
```

Comment:
Looks like the code is similar to:
```suggestion
        Tuple2<Map<TopicPartition, PartitionReassignmentState>, Boolean> t0 = findPartitionReassignmentStates(adminClient, targets);

        Map<TopicPartition, PartitionReassignmentState> partStates = t0.v1;
        Boolean partsOngoing = t0.v2;

        System.out.println(partitionReassignmentStatesToString(partStates));
        return new Tuple2<>(partStates, partsOngoing);
```
```suggestion
        Tuple2<Map<TopicPartition, PartitionReassignmentState>, Boolean> t0 = findPartitionReassignmentStates(adminClient, targets);

        System.out.println(partitionReassignmentStatesToString(t0.v1));

        return t0;
```

Reply:
Good catch. Thanks.